core/blocktree/blocktree_test.go

```diff
@@ -85,19 +86,52 @@ func createFlatTree(t *testing.T, depth int) *BlockTree {
 	return bt
 }
 
+func createFlatTree_WithWaitTime(t *testing.T, depth int) *BlockTree {
+	d := &db.BlockDB{
+		Db: db.NewMemDatabase(),
+	}
+
+	bt := NewBlockTreeFromGenesis(createGenesisBlock(), d)
+
+	previousHash := bt.head.Hash
+
+	for i := 1; i <= depth; i++ {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
For this case, I think you can eliminate the delay by manually adding the block. I also wonder if adding the timestamp as a parameter to `AddBlock` would make sense, eg. if we want to use time received.